libs/math/include/math/mat2.h

```diff
@@ -243,7 +243,9 @@ class MATH_EMPTY_BASES TMat22 :
         uint64_t result = 0;
         // For some reason clang is not able to vectoize this loop when the number of iteration
         // is known and constant (!?!?!). Still this is better than operator==.
+#if defined(__clang__)
```

Comment:
Is there an MSVC version of this?

Reply:
I think [Auto-Vectorization](https://docs.microsoft.com/en-us/cpp/parallel/auto-parallelization-and-auto-vectorization?view=msvc-170#auto-vectorizer) is the name of the similar MSVC feature. The exact SIMD solution it uses depends on `/arch` compiler flag (`/arch:x64` [defaults to SSE2](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-170) instruction set). But it is not as configurable as in GCC or Clang.

[hint_parallel(n)](https://docs.microsoft.com/en-us/cpp/preprocessor/loop) seems to be the closest I could find, but that's only available with an opt-in compiler flag I have never heard of before (`/Qpar`). Plus it belongs to Auto-**Parallelizer** feature which parallelizes code via **threads**, not via vectorization.

So my short answer is probably no, there isn't an MSVC version 😞

Reply:
sounds good thanks!